### DISCUSSION TITLE
정적 메서드는 어떤 경우에 사용하면 좋을 것 같나요? 단점은 없나요?

### DISCUSSION THREAD

**Discussion by wonsss** [woowacourse-study/2022-book-muscle#20] [마르코]
후이가 주신 질문이 좋아서, 디스커션으로 옮겼습니다.

[해당 PR](https://github.com/woowacourse-study/2022-book-muscle/pull/8 )

**Comment by wonsss** [comment]
https://github.com/woowacourse-study/2022-book-muscle/pull/8#discussion_r816164323

**Comment by juunzzi** [comment]
둘의 대화를 보고…

제가 저번 자동차 미션에서 정적메소드에 대해 궁금한게 정말 많았어요.. 리뷰어님 의견이 궁금해서 유틸성이 강하지만, 모듈(클래스)안에 있어야 할 법한 메소드들을 static으로 만들어 봤거든요 ?


돌아온 답변은

1. 너무 많은 static 키워드는 악이다 (이 부분은 마르코가 이해한 내용대로 저도 이해했던것같아요.)

2. 팩토리 메소드로 이용할 경우 사용할 수 있다.

아래는 참고한 링크입니다. 


https://velog.io/@godori/factory-method-pattern

https://tecoble.techcourse.co.kr/post/2020-07-16-static-method/
